### PR TITLE
Feature/keyvault/7.5

### PR DESCRIPTION
**Pull request by vcolin7**
Added `CertificateProperties.getX509ThumbprintString()` and updated service version to `7.5`.

Will add test recordings soon.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,18 @@
 # Release History
 
-## 4.6.0-beta.2 (Unreleased)
+## 4.6.0 (2024-02-20)
+Changes when compared to the last stable release (`4.5.9`) include:
 
 ### Features Added
+- Added support for service version `7.5`.
+- Added `CertificateProperties.getX509ThumbprintHexString()` to return the hexadecimal string representation of the SHA-1 hash of a certificate.
```
Comment (mccoyp) [inline sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md:8]: Was this property name chosen by Java's architects? .NET went with `X509ThumprintString`, and Python's still hashing out whether to use `string`, `str`, or `hex` as we speak

Comment (heaths) [reply sdk/keyvault/azure-security-keyvault-certificates/CHANGELOG.md:8]: A string representation is typically hex anyway, so this seems redundant. The most important thing is that it matches whatever ARM SDKs are using. @m-nash which ARM library needed this "X509ThumbprintString" again?

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java [hunk 2]
```diff
@@ -38,6 +38,26 @@ private static String getIdMetadataPiece(String[] pieces, int index) {
...
+            return null;
+        }
+
+        StringBuilder hexString = new StringBuilder();
+
+        for (byte b : x509Thumbprint) {
+            String hex = Integer.toHexString(0xFF & b);
+
+            if (hex.length() == 1) {
+                hexString.append('0');
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java:52]: Shouldn't you be prepending? I.e. likely an insert into index 0? But then that leads into my next comment below...

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java [hunk 3]
```diff
@@ -38,6 +38,26 @@ private static String getIdMetadataPiece(String[] pieces, int index) {
...
+            }
+
+            hexString.append(hex);
+        }
+
+        return hexString.toString().replace("-", "");
```
Comment (heaths) [inline sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java:58]: This is a lot of string manipulation both here and above. And why do you have dashes anyway? I didn't see where those are added above. You might see this in .NET often, but it's bad practice and lazy for .NET, which uses an older function that added them for some reason. I wrote my own for .NET, which is more common these days:

https://github.com/Azure/azure-sdk-for-net/blob/2dc0ca8a6a16025011fca42c671f0052bab922a0/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Extensions.cs#L16

That said, and as is now the case for newer .NET base64 encoding, it actually uses a preallocated table that is faster. That said, it takes more memory but if used frequently is better. This probably won't get used enough. But, basically, you create a 256-element array with its hex-encoded 2-byte string representation, then just index into that for each byte (as the index, that is).

**Review by heaths — APPROVED** [review]
A few non-blocking comments on hex encoding and naming, but otherwise LGTM.

File: sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/CertificatesUtils.java:58]: We have `CoreUtils.bytesToHexString` which could replace this entire method

**Review by alzimmermsft — APPROVED** [review]

**Review by heaths — APPROVED** [review]

**Review by heaths — APPROVED** [review]

**Pull request merged**